product_combination_exclude/models/product_product.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Graeme Gellatly
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+
+    _inherit = 'product.product'
+
+    @api.multi
+    def create(self, vals):
+        vals = dict(vals or {})
+        if 'attribute_value_ids' in vals:
+            exclusion_obj = self.env['product.attribute.exclude']
+            search_args = []
+            if 'product_tmpl_id' in vals:
+                search_args.extend([
```

Comment:
Not a biggie, but inplace operators are actually more efficient than `array.extend` due to no method call:
```
search_args += [
...
]
```